Sources/CoreCommands/SwiftTool.swift

```diff
@@ -780,7 +782,7 @@ public final class SwiftTool {
         if let sdk = options.build.customCompileSDK {
             swiftSDK.pathsConfiguration.sdkRootPath = sdk
         }
-        swiftSDK.architectures = options.build.architectures
+        swiftSDK.architectures = options.build.architectures.isEmpty ? nil : options.build.architectures
```

Comment:
I think we may want to reconsider whether the difference between empty array and `nil` is meaningful here. The fact that this worked without this suggests that it may not be.

BTW, this is so that `hostSwiftSDK == swiftSDK` can be true, right now it is always failing because `nil` is not equal to `[]`.